app/pb_kits/playbook/pb_source/source.rb

```diff
@@ -2,81 +2,46 @@
 
 module Playbook
   module PbSource
-    class Source < Playbook::PbKit::Base
-      PROPS = %i[configured_classname
-                 configured_data
-                 configured_hide_icon
-                 configured_id
-                 configured_source
-                 configured_type
-                 configured_user].freeze
+    class Source
+      include Playbook::Props
 
-      def initialize(classname: default_configuration,
-                     data: default_configuration,
-                     hide_icon: default_configuration,
-                     id: default_configuration,
-                     source: default_configuration,
-                     type: default_configuration,
-                     user: default_configuration)
-        self.configured_classname = classname
-        self.configured_data = data
-        self.configured_hide_icon = hide_icon
-        self.configured_id = id
-        self.configured_source = source
-        self.configured_type = type
-        self.configured_user = user
-      end
+      partial "pb_source/source"
 
-      def source
-        pb_source_title = Playbook::PbTitle::Title.new(size: 4, text: configured_source)
-        ApplicationController.renderer.render(partial: pb_source_title, as: :object)
-      end
+      prop :hide_icon, type: Playbook::Props::Boolean,
+                       default: false
+      prop :source, type: Playbook::Props::String
+      prop :type, type: Playbook::Props::Enum,
+                  values: %w[user retail inbound outbound prospecting events referral],
+                  default: "inbound"
+      prop :user, type: Playbook::Props::Hash, default: {}
 
       def type_text
-        pb_type = Playbook::PbBody::Body.new(color: "light") do
-          if type == "user" || (type == "referral" && is_set?(configured_user))
-            configured_user[:name]
-          else
-            type.titleize
-          end
+        if type == "user" || (type == "referral" && user)
+          user[:name]
+        else
+          type.titleize
         end
-        ApplicationController.renderer.render(partial: pb_type, as: :object)
       end
 
-      def type_icon
-        if !type_icon_name.nil? && avatar.nil?
-          pb_type_icon = Playbook::PbIconCircle::IconCircle.new(icon: type_icon_name, size: "sm")
-          ApplicationController.renderer.render(partial: pb_type_icon, as: :object)
-        end
+      def show_icon?
+        !type_icon_name.nil? && avatar.nil? ? true : false
       end
 
       def avatar
-        if is_set?(configured_user) && (type == "user" || type == "referral")
-          avatar_props = configured_user.clone
+        if user != {} && (type == "user" || type == "referral")
```

Comment:
Instead of `user != {}`, `!user.empty?` is a little more direct. If we add access to `ActiveSupport` in here, you could use `user.present?`

Reply:
One good article on these possibilities (there are many articles on this) - https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html